boxsdk/session/session.py

```diff
@@ -279,15 +279,29 @@ def _raise_on_unsuccessful_request(network_response: 'NetworkResponse', request:
                 context_info=response_json.get('context_info', None),
                 network_response=network_response
             )
+
+    @staticmethod
+    def _validate_json_response(network_response: 'NetworkResponse', request: '_BoxRequest', throw_exception=False) -> bool:
```

Comment:
it is a bad practice to design methods like that with a bool flag. `_validate_json_response` doesn't indicate that a bool will be a response and that exception can be thrown. I would create sth like:
```
def _is_response_format_valid(network_response: 'NetworkResponse', request: '_BoxRequest'):
    return request.expect_json_response and not is_json_response(network_response)
```
and keep throwing error in `_raise_on_unsuccessful_request`